workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowAbstractAction.java

```diff
@@ -0,0 +1,61 @@
+package com.symphony.bdk.workflow.management;
+
+import com.symphony.bdk.workflow.configuration.WorkflowDeployer;
+import com.symphony.bdk.workflow.exception.DuplicateException;
+import com.symphony.bdk.workflow.exception.NotFoundException;
+import com.symphony.bdk.workflow.swadl.SwadlParser;
+import com.symphony.bdk.workflow.swadl.v1.Workflow;
+
+import lombok.RequiredArgsConstructor;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+
+@RequiredArgsConstructor
+public abstract class WorkflowAbstractAction {
+
+  private final WorkflowDeployer deployer;
+
+  protected Workflow convertToWorkflow(String content) {
+    try {
+      return SwadlParser.fromYaml(content);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("SWADL content is not valid");
+    }
+  }
+
+  protected void validateFilePath(String id, String path) {
+    if (deployer.workflowSwadlPath(id) != null && deployer.workflowSwadlPath(id).endsWith(path)) {
+      throw new DuplicateException("SWADL file already exists");
+    }
+  }
+
+  protected Path getWorkflowFilePath(String id) {
+    return deployer.workflowSwadlPath(id);
+  }
+
+  protected boolean workflowExist(String id) {
+    return deployer.workflowExist(id);
+  }
+
+  protected void writeFile(String content, Workflow workflow, String path) {
+    try {
+      BufferedWriter writer = new BufferedWriter(new FileWriter(path, StandardCharsets.UTF_8));
+      writer.write(content);
+      writer.close();
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to write SWADL file " + workflow.getId());
```

Comment:
We can maybe tell more about the exception in the error message?

Reply:
👍  sure